### PR TITLE
Date range validation

### PR DESCRIPTION
**Pull request by joanico**
Closes #issue_number

## validate for End date must be greater than the start date
- 
## Instructions on making this work
Screen Shot 2021-10-01 at 2 32 52 PM

### CODE REVIEW

File: src/components/AddIstoria.vue [hunk 1]
```diff
@@ -51,6 +58,14 @@
     import { mapActions } from 'vuex'
 
     export default {
+        name: "Submited",
```
Comment (PeteCoward) [inline src/components/AddIstoria.vue:61]:
```suggestion
```

????

File: src/components/AddIstoria.vue [hunk 2]
```diff
@@ -1,6 +1,5 @@
 <template>
     <div class="card mb-3">
-        {{ add_istoria }}
         <div class="card-body">
             <!-- Default form -->
             <form v-if="amILoggedIn === true" @submit.prevent="submitNewJourney" >
```
Comment (PeteCoward) [inline src/components/AddIstoria.vue:5]:
```suggestion
            <form v-if="amILoggedIn === true" @submit.prevent="validateAndSubmitNewJourney" >
```

File: src/components/AddIstoria.vue [hunk 3]
```diff
@@ -60,6 +75,17 @@
         },
         methods: {
             ...mapActions(['detectLogin', 'submitNewJourney']),
+            submitNewJourney(e){
```
Comment (PeteCoward) [inline src/components/AddIstoria.vue:78]:
```suggestion
            validateAndSubmitNewJourney(e){
```

File: src/components/AddIstoria.vue [hunk 4]
```diff
@@ -60,6 +75,17 @@
...
+                if(this.fromDate && this.toDate){
+                    console.log("submit function called")
+                }
+                this.error=[];
+                if(this.fromDate > this.toDate){
+                    this.error.push("End date must be greater than start date")
+                }
+                console.warn("error", this.error)
+                e.preventDefault()
```
Comment (PeteCoward) [inline src/components/AddIstoria.vue:87]:
```suggestion
                if ( this.error.length === 0 ) {
                    this.submitNewJourney(e);
                }
```

File: src/components/AddIstoria.vue [hunk 5]
```diff
@@ -51,6 +58,13 @@
     import { mapActions } from 'vuex'
 
     export default {
+        data(){
+            return {
+                error:[],
```
Comment (eraulo) [inline src/components/AddIstoria.vue:63]:
```suggestion
                error:"",
```

File: src/components/AddIstoria.vue [hunk 6]
```diff
@@ -60,6 +74,18 @@
         },
         methods: {
             ...mapActions(['detectLogin', 'submitNewJourney']),
+            validateAndSubmitNewJourney(e){
+                if(this.fromDate && this.toDate){
+                    console.log("submit function called")
+                }
+                this.error=[];
```
Comment (eraulo) [inline src/components/AddIstoria.vue:81]:
```suggestion
```

File: src/components/AddIstoria.vue [hunk 7]
```diff
@@ -14,9 +13,17 @@
...
                         <label for="toDate">to</label>
-                        <input type="date" class="toDate" name="toDate" select=":last" required="" style="width: 400px; margin: 8px;" />
+                        <input type="date" class="toDate" name="toDate" select=":last" required="" v-model="toDate" style="width: 400px; margin: 8px;" />
+                        <p v-if="error.length" style="color: red;">
+                            <b>Please correct following this error:</b>
+                            <ul>
+                                <li v-for="e in error" v-bind:key="e.id">
+                                    {{e}}
+                                </li>
+                            </ul>
```
Comment (eraulo) [inline src/components/AddIstoria.vue:25]:
```suggestion
<p>{{ error }}</p>
                              
```

File: src/components/AddIstoria.vue [hunk 8]
```diff
@@ -60,6 +74,18 @@
         },
         methods: {
             ...mapActions(['detectLogin', 'submitNewJourney']),
+            validateAndSubmitNewJourney(e){
+                if(this.fromDate && this.toDate){
+                    console.log("submit function called")
+                }
+                this.error=[];
+                if(this.fromDate > this.toDate){
+                    this.error.push("End date must be greater than start date")
```
Comment (eraulo) [inline src/components/AddIstoria.vue:83]:
```suggestion
                    this.error += "End date must be greater than start date"
```

Comment (joanico) [reply src/components/AddIstoria.vue:83]: @eraulo if I change according to your suggestion the result is as follows:
Screen Shot 2021-09-30 at 8 04 49 PM

File: src/components/AddIstoria.vue [hunk 9]
```diff
@@ -14,9 +13,17 @@
                     </div>
                     <div class="form-group"> <!-- Date input -->
                         <label for="fromDate">From </label>
-                        <input type="date" class="fromDate" name="fromDate" select=":first" required="" style="width: 400px; margin: 8px;" />
+                        <input type="date" class="fromDate" name="fromDate" select=":first" required="" v-model="fromDate" style="width: 400px; margin: 8px;" />
                         <label for="toDate">to</label>
-                        <input type="date" class="toDate" name="toDate" select=":last" required="" style="width: 400px; margin: 8px;" />
+                        <input type="date" class="toDate" name="toDate" select=":last" required="" v-model="toDate" style="width: 400px; margin: 8px;" />
+                        <p v-if="error.length" style="color: red;">
```
Comment (PeteCoward) [inline src/components/AddIstoria.vue:19]:
```suggestion
                        <p v-if="errors.length" style="color: red;">
```

File: src/components/AddIstoria.vue [hunk 10]
```diff
@@ -14,9 +13,17 @@
...
+                            <b>Please correct following this error:</b>
+                            <ul>
+                                <li v-for="e in error" v-bind:key="e.id">
```
Comment (PeteCoward) [inline src/components/AddIstoria.vue:22]:
```suggestion
                                <li v-for="e in errors" v-bind:key="e.id">
```

File: src/components/AddIstoria.vue (hunk 5 above)
Comment (PeteCoward) [inline src/components/AddIstoria.vue:63]:
```suggestion
                errors: [],
```

File: src/components/AddIstoria.vue (hunk 6 above)
Comment (PeteCoward) [inline src/components/AddIstoria.vue:81]:
```suggestion
                this.errors = [];
```

File: src/components/AddIstoria.vue [hunk 11]
```diff
@@ -60,6 +74,18 @@
         },
         methods: {
             ...mapActions(['detectLogin', 'submitNewJourney']),
+            validateAndSubmitNewJourney(e){
+                if(this.fromDate && this.toDate){
+                    console.log("submit function called")
+                }
```
Comment (PeteCoward) [inline src/components/AddIstoria.vue:80]: maybe remove this code?

File: src/components/AddIstoria.vue [hunk 12]
```diff
@@ -60,6 +74,18 @@
...
+                this.error=[];
+                if(this.fromDate > this.toDate){
+                    this.error.push("End date must be greater than start date")
+                }
+                if( this.error.length === 0 ) {
```
Comment (PeteCoward) [inline src/components/AddIstoria.vue:85]:
```suggestion
                if( this.errors.length === 0 ) {
```

File: src/components/AddIstoria.vue (hunk 8 above)
Comment (PeteCoward) [inline src/components/AddIstoria.vue:83]:
```suggestion
                    this.errors.push("End date must be greater than start date")
```

**Review by PeteCoward — CHANGES_REQUESTED** [review]
please rename error to errors, and `git merge origin/master`

**Review by PeteCoward — APPROVED** [review]

**Pull request merged**